src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralReferral.java

```diff
@@ -95,4 +96,22 @@
         typeOverride = TextArea
     )
     private String generalReferralDecisionReason;
+
+    @JsonIgnore
+    public GeneralReferral copy() {
+        return GeneralReferral.builder()
+            .generalReferralReason(generalReferralReason)
+            .generalApplicationFrom(generalApplicationFrom)
+            .generalApplicationReferralDate(generalApplicationReferralDate)
+            .generalApplicationAddedDate(generalApplicationAddedDate)
+            .generalReferralType(generalReferralType)
+            .alternativeServiceMedium(alternativeServiceMedium)
+            .generalReferralJudgeOrLegalAdvisorDetails(generalReferralJudgeOrLegalAdvisorDetails)
+            .generalReferralFeeRequired(generalReferralFeeRequired)
+            .generalReferralFee(generalReferralFee)
+            .generalReferralDecision(generalReferralDecision)
+            .generalReferralDecisionDate(generalReferralDecisionDate)
+            .generalReferralDecisionReason(generalReferralDecisionReason)
+            .build();
```

Comment:
can we not use `@Builder(toBuilder=true)` on GeneralReferral and get a copy of it ?

Reply:
Update in this commit https://github.com/hmcts/nfdiv-case-api/pull/1269/commits/e0baf9e5588269dde2f610f2464e6d1ed66ffe45